sqlserver/datadog_checks/sqlserver/connection.py

```diff
@@ -28,6 +28,81 @@ class SQLConnectionError(Exception):
     pass
 
 
+# we're not including space in the set of invalid characters because it's a valid key character for adodbapi
+# best be more permissive here so that the driver and the database have the final say as to whether it's valid or not
+CONNECTION_STRING_SPECIAL_CHARACTERS = set('=;[]{}"\'')
+
+
+def parse_connection_string_properties(cs):
+    """
+    Parses the properties portion of a SQL Server connection string (i.e. "key1=value1;key2=value2;...") into a map of
+    {key -> value}. The string must contain *properties only*, meaning the subprotocol, serverName, instanceName and
+    portNumber are not included in the string.
+    See https://docs.microsoft.com/en-us/sql/connect/jdbc/building-the-connection-url
+    """
+    cs = cs.strip()
+    params = {}
+    i = 0
+    escaping = False
+    key, parsed, key_done = "", "", False
+    while i < len(cs):
+        if escaping:
+            if cs[i : i + 2] == '}}':
+                parsed += '}'
+                i += 2
+                continue
+            if cs[i] == '}':
+                escaping = False
+                i += 1
+                continue
+            parsed += cs[i]
+            i += 1
+            continue
+        if cs[i] == '{':
```

Comment:
It could, but I don't think that's more readable. With `continue` it's clear that the flow is ending right there and going back to the top. With a really long if-elif you have to keep reading to see if there's more stuff that happens at the end.